src/main/scala/com/databricks/labs/overwatch/pipeline/Restore.scala

```diff
@@ -0,0 +1,126 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.env.{Database, Workspace}
+import com.databricks.labs.overwatch.utils.Helpers.{logger, removeTrailingSlashes}
+import com.databricks.labs.overwatch.utils.{BadConfigException, CloneDetail, Config, Helpers, SparkSessionWrapper, WriteMode}
+import org.apache.log4j.{Level, Logger}
+import com.databricks.labs.overwatch.pipeline.Snapshot
+
+import java.io.FileNotFoundException
+
+
+/**
+ *
+ * @param _sourceETLDB        ETL Database for Souce from where Restore need to be done.
+ * @param _targetPrefix       Target Path for where Restore would be done
+ * @param _workspace          Workspace from where Restore would be performed
+ * @param _database           Workspace Database Name.
+ * @param _config             Source Workspace Config.
+ */
+
+class Restore (_sourceETLDB: String, _targetPrefix: String, _workspace: Workspace, _database: Database, _config: Config)
+  extends Pipeline(_workspace, _database, _config) {
+
+  import spark.implicits._
+
+  private val targetPrefix = removeTrailingSlashes(_targetPrefix)
+  private val workSpace = _workspace
+  private val sourceETLDB = _sourceETLDB
+  private val bronze = Bronze(workSpace)
+  private val silver = Silver(workSpace)
+  private val gold = Gold(workSpace)
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  val Config = _config
+
+  private[overwatch] def restore(
+                                  allTarget: Array[PipelineTable],
+                                  cloneLevel: String = "DEEP",
+                                ): Unit = {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
```

Comment:
These should be an enum that you use consistently across Snapshot, Restore and Migration, as needed, so that you remove the option for user error